### ISSUE TITLE
ruby gems periodically fail to load as expected

### ISSUE THREAD

**Issue by macb** [drone/drone#1349]
This was a new intermittent build failure seen when switching to drone 0.4. The projects were mostly untouched aside from switching to drone 0.4 and bits required to make that switch possible.

A sample drone yaml looks like:
```
build:
  image: internal.docker.index/ruby:latest
  pull: true
  environment:
    - DATABASE_MASTER_HOST=127.0.0.1
    - DATABASE_USER=root
    - DATABASE_PASSWORD=
  commands:
    - bundle install --path=bundler
    - script/cibuild
cache:
  mount:
    - bundler
compose:
  database:
    image: mysql:5.5
    environment:
      - MYSQL_ALLOW_EMPTY_PASSWORD="yes"
```

The internal ruby image is based on the community ruby image with additional utilities added (like `mysql-client-5.5`). With the build failing when attempting to run `script/cibuild` (which executes `bundle exec rspec` after setting up the DBs.). The output looks something like:

```
/drone/src/github.com/company/application/config/application.rb:114:in `require': cannot load such file -- foo/bar/baz (LoadError)
```

The changes made to make this work with drone 0.4 was changing the `bundle install --path` to be local to the direct (we've also tried `/drone/bundler`) to facilitate the caching plugin's requirements. In another project where we don't bundle into the `/drone` directory or a child directory I don't recall seeing these load errors.

I'm not entirely convinced this is a problem with drone itself and likely more to do with docker and any volumes involved but it's showing up as build failures currently. I'm working to create a test application to try and reproduce the failure in an application I can share publicly.

**Comment by macb** [comment]
https://github.com/macb/rspec-rails-examples should serve as a reasonable rails application example. I haven't gotten a load error issue with it yet (been running with `drone exec` locally) but maybe @bradrydzewski can see it fail.

**Comment by bradrydzewski** [comment]
I think this issue has been fixed. I modified the cache plugin to disable tarball compression and then I was unable to repeat the error after 60 successful attempts. See https://github.com/drone-plugins/drone-cache/pull/4

I also updated the plugin to delete corrupted tarballs. This may still result in a build failure, but subsequent builds will no longer be impacted by the corrupted tarball, preventing the infinite loop of failures.

I'm hoping this issue can be closed for good, but we can re-open if you continue to observe issues.
